discovery-provider/src/queries/get_trending_tracks.py

```diff
@@ -59,9 +60,29 @@ def generate_unpopulated_trending(
         for track in trending_tracks["listen_counts"]
     ]
 
+    # If usdc_purchase_only is true, then filter out track ids belonging to
```

Comment:
tbh i don't love how this is combined with the trending code. i'm cool with moving forward, but i think we do want to start peeling apart API & query layers from complicating eachother.